### PR TITLE
[DSM] DDP-7234: user_changes

### PR DESCRIPTION
**Pull request by pegahtah**

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/roles/RoleDao.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.dao.Dao;
+import org.broadinstitute.dsm.db.jdbi.JdbiRole;
+import org.broadinstitute.dsm.db.jdbi.RoleDto;
+import org.broadinstitute.lddp.db.SimpleResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RoleDao implements Dao<RoleDto> {
+    Logger logger = LoggerFactory.getLogger(RoleDao.class);
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/roles/RoleDao.java]: Let's use @Slf4j annotation instead

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java [hunk 2]
```diff
@@ -688,6 +697,25 @@ public Object handle(Request request, Response response) throws Exception {
         logger.info("Finished setting up DSM custom routes and jobs...");
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java]: Let's replace logger with injected one using @Slf4j annotation

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/user/UserDao.java [hunk 3]
```diff
@@ -5,19 +5,26 @@
...
+import org.broadinstitute.dsm.exception.DaoException;
+import org.broadinstitute.dsm.util.DBUtil;
 import org.broadinstitute.lddp.db.SimpleResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class UserDao implements Dao<UserDto> {
-
-    long EXPIRATION_DURATION_MILLIS = TimeUnit.HOURS.toMillis(24);
+    Logger logger = LoggerFactory.getLogger(UserDao.class);
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/user/UserDao.java]: Let's use @Slf4j annotation instead

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/user/RoleDto.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package org.broadinstitute.dsm.db.dto.user;
+
+import com.google.gson.annotations.SerializedName;
+import lombok.Data;
+
+@Data
+public class RoleDto {
+    @SerializedName ("name")
+    String roleName;
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/user/RoleDto.java]: Maybe it's better to rename the column from roleName to name?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/user/RoleDto.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    long roleId;
+    long umbrellaId;
+    String description;
+
+    public RoleDto(String roleName, long roleId, String description, long umbrellaId) {
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/user/RoleDto.java]: Could be replaced with AllArgsConstructor annotation

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/user/UserDto.java [hunk 6]
```diff
@@ -2,28 +2,34 @@
...
     private String firstName;
     private String lastName;
     private String name;
     private String email;
     private String phoneNumber;
     private String auth0UserId;
+    private boolean isActive;
 
-    public UserDto(long userId, String name, String email, String phoneNumber, String auth0UserId) {
+    public UserDto(long userId, String name, String email, String phoneNumber, String auth0UserId, String guid, String firstName,
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/user/UserDto.java]: Same here. AllArgsConstructor annotation will work

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/jdbi/RoleDto.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.Data;
+
+@Data
+public class RoleDto {
+    long roleId;
+    String name;
+    String description;
+    long umbrellaId;
+
+    public RoleDto(long roleId, String name, String description, long umbrellaId) {
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/jdbi/RoleDto.java]: AllArgsConstructor

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/access/GetRoleRoute.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import org.broadinstitute.dsm.util.UserUtil;
+import org.broadinstitute.lddp.handlers.util.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+public class GetRoleRoute extends RequestHandler {
+    private Logger logger = LoggerFactory.getLogger(GetRoleRoute.class);
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/access/GetRoleRoute.java]: Let's use @Slf4j annotation instead

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/GetUserRoute.java [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+import org.broadinstitute.dsm.util.UserUtil;
+import org.broadinstitute.lddp.handlers.util.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+public class GetUserRoute extends RequestHandler {
+    private static Logger logger = LoggerFactory.getLogger(GetUserRoute.class);
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/GetUserRoute.java]: Let's use @Slf4j annotation instead

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/GetUserRoute.java [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    private UserRoleDao userRoleDao;
+
+    public GetUserRoute(UserRoleDao userRoleDao) {
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/GetUserRoute.java]: AllArgsConstructor

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/ModifyUserRoute.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+import org.broadinstitute.lddp.handlers.util.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+public class ModifyUserRoute extends RequestHandler {
+
+    Logger logger = LoggerFactory.getLogger(ModifyUserRoute.class);
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/ModifyUserRoute.java]: Let's use @Slf4j annotation instead

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/ModifyUserRoute.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    UserRoleDao userRoleDao;
+
+    public ModifyUserRoute(UserRoleDao userRoleDao) {
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/ModifyUserRoute.java]: AllArgsConstructor

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/PostUserRoute.java [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+import org.broadinstitute.dsm.util.UserUtil;
+import org.broadinstitute.lddp.handlers.util.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+public class PostUserRoute extends RequestHandler {
+    private static Logger logger = LoggerFactory.getLogger(PostUserRoute.class);
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/PostUserRoute.java]: Let's use @Slf4j annotation instead

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/PostUserRoute.java [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+    String auth0Account;
+    String clientKey;
+    private UserDao userDao;
+
+    public PostUserRoute(UserDao userDao, String auth0Account, String clientKey) {
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/PostUserRoute.java]: AllArgsConstructor

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/patch/DeactivateUserRoute.java [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+import org.broadinstitute.lddp.handlers.util.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+public class DeactivateUserRoute extends RequestHandler {
+
+    Logger logger = LoggerFactory.getLogger(ModifyUserRoute.class);
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/patch/DeactivateUserRoute.java]: Let's use @Slf4j annotation instead

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/patch/DeactivateUserRoute.java [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+    UserDao userDao;
+
+    public DeactivateUserRoute(UserDao userDao) {
```
Comment (Trimax) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/patch/DeactivateUserRoute.java]: AllArgsConstructor

**Comment by Trimax** [comment]
General comment: not sure if both DSS & DSM teams will use the same Checkstyle configuration but this code obviously doesn't follow the one that we're using (for instance, spaces between brackets and annotations, empty lines, etc).

**Comment by pegahtah** [comment]
> General comment: not sure if both DSS & DSM teams will use the same Checkstyle configuration but this code obviously doesn't follow the one that we're using (for instance, spaces between brackets and annotations, empty lines, etc).

Yes I haven't done the check style checks, I will do them. I meant to change this to draft PR for some time

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/patch/DeactivateUserRoute.java (hunk 15 above)
Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/user/patch/DeactivateUserRoute.java]: Thanks!! I will

**Pull request merged**